src/test/java/core/basesyntax/operation/PurchaseOperationHandlerTest.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.FruitTransactionException;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operation.OperationHandler;
+import core.basesyntax.service.operation.PurchaseOperationHandler;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class PurchaseOperationHandlerTest {
+    private static final String VALID_FRUIT_NAME = "banana";
+    private static final Integer VALID_FRUIT_AMOUNT = 10;
+    private static final String OPERATION_CODE = "p";
+    private static final Integer ZERO_FRUIT_AMOUNT = 0;
+    private OperationHandler operationHandler;
+
+    @BeforeEach
+    void setUp() {
+        operationHandler = new PurchaseOperationHandler();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.STORAGE.clear();
+    }
+
+    @Test
+    void executeOperation_ValidData_ok() {
+        Storage.STORAGE.put(VALID_FRUIT_NAME, VALID_FRUIT_AMOUNT);
```

Comment:
better to make test in this way: 1.Precondition 2. Executioin 3. Assertions. And each step separate by empty line